### PR TITLE
chore: returns on utils,queue,cosign

### LINKED ISSUE #1526: missing return statements on queue, utils, cosign
### Describe what should be investigated or refactored

As we create uniforn typing standards across our projects, we need to return types on queue, cosign, and utils

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

As we continue to strictly type, this is the next map of files that need return types to improve our typing system

## Related Issue

Fixes #1526 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: src/lib/queue.ts [hunk 1]
```diff
@@ -73,7 +81,7 @@ export class Queue<K extends KubernetesObject> {
    *
    * @returns A promise that resolves when the webapp is reconciled
    */
-  async #dequeue() {
+  async #dequeue(): Promise<false | undefined> {
```
Comment (samayer12) [inline src/lib/queue.ts:84]: returning `false | undefined` makes me wonder about where this is consumed. There's the possibility for some unexpected behavior in conditional statements.
> `false`: Is falsy but has a defined type and value.
> `undefined`: Is also falsy but indicates the absence of a value.

Do you think this justifies a small follow-up issue to dig into how this is consumed? It's definitely out-of-scope on this PR focused on just _adding_ return types.

**Review by samayer12 — APPROVED** [review]
typing++, thanks!

File: src/lib/queue.ts (hunk 1 above)
Comment (cmwylie19) [reply src/lib/queue.ts:84]: as the queue is such a critical piece i definitely think it deserves a close look

Comment (cmwylie19) [reply src/lib/queue.ts:84]: https://github.com/defenseunicorns/pepr/issues/1530

**Pull request merged**